pkg/cmd/root/help_reference.go

```diff
@@ -19,7 +19,11 @@ func referenceHelpFn(io *iostreams.IOStreams) func(*cobra.Command, []string) {
 			wrapWidth = io.TerminalWidth()
 		}
 
-		md, err := markdown.Render(cmd.Long,
+		// Assumes that we want to produce reference text for the entire tree of commands,
+		// starting from the root.
+		referenceText := stringifyCmdReference(cmd.Root())
```

Comment:
This is really the difference maker, by pushing the stringification into the help func, we know that this command has been attached somewhere below the root (at least, if my understanding of `cmd.Root()` is correct 🌚 ) so we can walk the command tree.

Note that doing this has two consequences:
 * The `cmd.Long` is not set (but I'm not sure if this was used except to serve as a place to store the markdown to render)
 * Markdown rendering would occur each time we call the help func (but I can't think of a reason it would be called multiple times in one CLI invocation)

Reply:
The reason it is important to set `cmd.Long` is for the auto generating of our [online manual](https://cli.github.com/manual/gh_help_reference). This gets generated at release time [here](https://github.com/cli/cli/blob/b01c27aba251c25353b570142d998cfa0ba2c00c/.github/workflows/releases.yml#L65) which eventually calls [this file](https://github.com/cli/cli/blob/trunk/cmd/gen-docs/main.go). This is not at all obvious from the code and we have fielded questions about how the online docs are generated multiple times. We will need to figure out how to best solve this so that the online manual is not effected.

Reply:
Nice! This is exactly the kind of thing I was hoping to get out of this PR. I'll look into it and see what I can do.

Reply:
It's kind of unfortunate because then we'd diverge on real CLI usage vs the doc generation to produce this, which is maybe more complicated than an explicit comment. Explicit and a bit janky is better than implicit and adding layers of abstraction to make it _seem_ less janky.

I'm inclined to back out the `reference` specific work and just leave the structural changes in place with an additional comment on what we've discussed here. Does that sound reasonable?

Reply:
I also don't feel like it's very important to remove the temporal coupling of the `reference` command if it's going to be too large of a pull, especially because there is already a comment explaining why that specific line should remain last. Feel free to reduce the scope of this work if you would.

Reply:
Thank you for any feedback